### PR TITLE
Upgrade to `egui@0.29.1` and `winit@0.30.5`

### PR DESCRIPTION
**Pull request by Gae24**
This is the minimum amount of work necessary to upgrade to `egui@0.29.1`. Updating both `glow` and `winit` were required. Various deprecated `winit` methods are still used; the switch to the trait based API will be done in a followup PR. `egui-glow` has added a boolean on the `Painter` constructor to enable dithering, which for now is set to `false`. `epaint` has a new dependency `epaint_default_fonts` for the `default_fonts` feature.

---

- [X] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [X] These changes do not require tests because they do not change functionality.

### CODE REVIEW

File: servo-tidy.toml [hunk 1]
```diff
@@ -52,9 +52,11 @@ packages = [
     "foreign-types",
     "foreign-types-shared",
     "metal",
-    "glow",
     "windows-core",
 
+    # wgpu-hal depends on 0.5.0
```
Comment (mrobinson) [inline servo-tidy.toml:57]:
```suggestion
    # wgpu-hal depends on 0.5.0.
```

File: ports/servoshell/desktop/minibrowser.rs [hunk 2]
```diff
@@ -78,7 +78,7 @@ fn truncate_with_ellipsis(input: &str, max_length: usize) -> String {
 impl Minibrowser {
     pub fn new(
         rendering_context: &RenderingContext,
-        events_loop: &EventsLoop,
+        event_loop: &winit::event_loop::EventLoop<WakerEvent>,
```
Comment (mrobinson) [inline ports/servoshell/desktop/minibrowser.rs:81]: It's a bit odd to only used a qualified use for one of these types, so I would also do the same for `EventLoop`.

```suggestion
        event_loop: &EventLoop<WakerEvent>,
```

File: ports/servoshell/desktop/headed_window.rs [hunk 3]
```diff
@@ -72,15 +72,15 @@ impl Window {
         // #9996.
         let visible = opts.output_file.is_none() && !no_native_titlebar;
 
-        let window_builder = winit::window::WindowBuilder::new()
+        let window_attr = winit::window::Window::default_attributes()
```
Comment (mrobinson) [inline ports/servoshell/desktop/headed_window.rs:75]:
```suggestion
        let window_attributes = winit::window::Window::default_attributes()
```

File: ports/servoshell/desktop/events_loop.rs [hunk 4]
```diff
@@ -62,7 +63,7 @@ impl EventsLoop {
         Ok(EventsLoop(if headless {
             EventLoop::Headless(Arc::new((Mutex::new(false), Condvar::new())))
         } else {
-            let mut event_loop_builder = winit::event_loop::EventLoopBuilder::with_user_event();
+            let mut event_loop_builder = winit::event_loop::EventLoop::with_user_event();
```
Comment (mrobinson) [inline ports/servoshell/desktop/events_loop.rs:66]: Can you also do a use `winit::event_loop::EventLoop` above and replace all the of the qualified usage in this file? It's odd to mix them. I know that the existing code does this, but we should start cleaning it up.

```suggestion
            let mut event_loop_builder = EventLoop::with_user_event();
```

**Review by mrobinson — CHANGES_REQUESTED** [review]
Thank you! Looks good with a couple minor nits.

File: ports/servoshell/desktop/events_loop.rs (hunk 4 above)
Comment (Gae24) [reply ports/servoshell/desktop/events_loop.rs:66]: I guess it was done because there's already the `EventLoop` enum in scope

**Review by mrobinson — APPROVED** [review]

**Pull request merged**